data/src/main/java/org/apache/iceberg/data/BaseDeleteLoader.java

```diff
@@ -146,6 +151,26 @@ private <T> Iterable<T> materialize(CloseableIterable<T> iterable) {
   @Override
   public PositionDeleteIndex loadPositionDeletes(
       Iterable<DeleteFile> deleteFiles, CharSequence filePath) {
+    if (containsDVs(deleteFiles)) {
+      DeleteFile dv = Iterables.getOnlyElement(deleteFiles);
+      validateDV(dv, filePath);
+      return readDV(dv); // TODO: support caching entire DV files
```

Comment:
A bit of context about how caching works for V2 deletes. If we estimate the content of entire file to fit into the cache (its in-memory representation), we read the entire file and cache bitmaps for each data file we read. We can do similar stuff for Puffin. I need to explore the performance impact of not knowing the footer size upfront.

Any early feedback is welcome!

Reply:
I am also not sure it should be a blocker.

Reply:
I am not sure we can efficiently implement caching with the current Puffin reader, as it would require at least 3 requests (one to find the footer size, one to read the footer, and one for the content). In theory, we can do this in a single read, given that we know the overall size of the Puffin file. Therefore, I suggest that we keep this TODO and revisit it later.